Content.Server/Lathe/LatheSystem.cs

```diff
@@ -149,7 +151,8 @@ public bool TryStartProducing(EntityUid uid, LatheComponent? component = null)
             var recipe = component.Queue.First();
             component.Queue.RemoveAt(0);
 
-            EnsureComp<LatheProducingComponent>(uid);
+            var lathe = EnsureComp<LatheProducingComponent>(uid);
+            lathe.ProductionLength = (float) recipe.CompleteTime.TotalSeconds * component.SpeedMultiplier;
```

Comment:
The math is wrong here. A higher speed multiplier actually makes it slower.

Reply:
You sure? The multiplier is calculated with `modifier ^ (rating - 1)`, so it should be equal to 1 at rating 1 and go down from there. Then that decimal is simply multiplied by the time it takes to print the recipe. Testing it in game seemed to work fine as well. Perhaps the variable just has a confusing name?

Reply:
If SpeedMultiplier is meant to multiply production **speed** rather than production **time** like the name suggests, then PartRatingPrintSpeedMultiplier should be 2 instead of 0.5 so the multiplier goes up when upgraded, and the production time math should be changed so it divides the total production time by SpeedMultiplier, which is the same as multiplying speed.

Reply:
I meant production time all along. whoops! renamed and pushed to clear that up.